packages/java-edition/src/index.ts

```diff
@@ -116,6 +118,59 @@ export const initialize: core.ProjectInitializer = async (ctx) => {
 				!n.symbol?.path[0]?.startsWith('::minecraft')),
 	})
 
+	mcdoc.runtime.registerAttribute(
+		meta,
+		'since',
+		mcdoc.runtime.attribute.validator.string,
+		{
+			filterElement: (config, ctx) => {
+				if (!config.startsWith('1.')) {
+					ctx.logger.warn(`Invalid mcdoc attribute for "since": ${config}`)
+					return true
+				}
+				return ReleaseVersion.cmp(release, config as ReleaseVersion) >= 0
+			},
+		},
+	)
+	mcdoc.runtime.registerAttribute(
+		meta,
+		'until',
+		mcdoc.runtime.attribute.validator.string,
+		{
+			filterElement: (config, ctx) => {
+				if (!config.startsWith('1.')) {
+					ctx.logger.warn(`Invalid mcdoc attribute for "until": ${config}`)
+					return true
+				}
+				return ReleaseVersion.cmp(release, config as ReleaseVersion) < 0
+			},
+		},
+	)
+	mcdoc.runtime.registerAttribute(
+		meta,
+		'deprecated',
```

Comment:
When is this used instead of `since` / `until`?

Reply:
`show_item` hover event has some of these

Reply:
Ah, so it's essentially when it's still valid in vanilla mc but deprecated?